station-dispenser-api-v0/src/main/java/net/modificationstation/stationapi/api/item/DispenseUtil.java

```diff
@@ -0,0 +1,132 @@
+package net.modificationstation.stationapi.api.item;
+
+import net.minecraft.block.entity.DispenserBlockEntity;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.ItemEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.World;
+
+import java.util.Random;
+
+/**
+ * Utility class for changing dispenser behavior, providing tools to customize behavior
+ * @author matthewperiut
+ * @see DispenseEvent
+ * @see CustomDispenseBehavior
+ */
+public class DispenseUtil {
+    public World world;
+    public DispenserBlockEntity dispenserBlockEntity;
+    public ItemStack itemStack;
+    private ItemStack shotItemStack;
+    public final ItemStack[] inventory;
+    public final int slot;
+    public Random random = new Random();
+    public int x;
+    public int y;
+    public int z;
+    public byte xDir = 0;
+    public byte zDir = 0;
+    public double xVel;
+    public double yVel;
+    public double zVel;
+
+    public DispenseUtil(World world, ItemStack itemStack, DispenserBlockEntity dispenserBlockEntity, ItemStack[] inventory, int slot) {
+        this.world = world;
+        this.dispenserBlockEntity = dispenserBlockEntity;
+        this.inventory = inventory;
+        this.itemStack = itemStack;
+        this.slot = slot;
+        populateData();
+    }
+
+    public void decrementItem(int amount) {
+        if (inventory != null) {
+            if (inventory[slot] != null) {
+                if (inventory[slot].count - amount <= 0) {
+                    inventory[slot] = null;
+                } else {
+                    shotItemStack = inventory[slot].split(amount);
+                }
+                dispenserBlockEntity.markDirty();
+            }
+        }
+    }
+
+    public void setItem(ItemStack itemStack) {
+        if (inventory != null) {
+            inventory[slot] = itemStack;
+            dispenserBlockEntity.markDirty();
+        }
+    }
+
+    private void populateData() {
+        x = dispenserBlockEntity.x;
+        y = dispenserBlockEntity.y;
+        z = dispenserBlockEntity.z;
+        int meta = world.getBlockMeta(x, y, z);
+        if (meta == 3) {
+            zDir = 1;
+        } else if (meta == 2) {
+            zDir = -1;
+        } else if (meta == 5) {
+            xDir = 1;
+        } else {
+            xDir = -1;
+        }
+        xVel = x + xDir * 0.6 + 0.5;
+        yVel = y + 0.5;
+        zVel = z + zDir * 0.6 + 0.5;
+    }
+
+    public void shootItemStack() {
```

Comment:
`shootStack` works better with yarn names

Reply:
shootStack it is!